src/ucp/core/ucp_context.h

```diff
@@ -13,7 +13,77 @@
 #include <uct/api/uct.h>
 #include <ucs/datastruct/queue_types.h>
 #include <ucs/type/component.h>
-
+#include <ucs/type/spinlock.h>
+#include "config.h"
+
+typedef struct ucp_mt_config {
+    int                           mt_required;
+    int                           use_mutex;
+    union {
+        /* Lock for multithreading support. Either spinlock or mutex is used at
+           at one time. Spinlock is the default option. */
+        pthread_mutex_t           mt_mutex;
+        ucs_spinlock_t            mt_spinlock;
+    } lock;
+} ucp_mt_config_t;
+
+extern int              num_contexts;
+extern ucp_mt_config_t  mm_mt_config;
+
+#if ENABLE_MT
+#  define UCP_THREAD_LOCK_INIT(_config)                                 \
```

Comment:
_config parameter to these macros should be a pointer to config, because it's being modified by the function